### DISCUSSION TITLE
PGRST102 for postman BASH INSERT A ROW

### DISCUSSION THREAD

**Discussion by gflegg** [supabase/supabase#10461] [Questions] [answered]
Why, Following the supabase crash course, using postman to READ ALL ROWS and INSERT A ROW:
-  is there no problem for READ ALL ROWS
- a problem for INSERT A ROW as follows 
{
"code": "PGRST102",
    "details": null,
    "hint": null,
    "message": "Error in $: string at 'name=item03+test&='"
}
<img width="2152" alt="2022-11-23 (2)" src="https://user-images.githubusercontent.com/71608873/203336213-ba75fcc3-52c5-4f07-aa8a-a7b5492a38e6.png">

**Comment by GaryAustin1** [comment, marked as answer]
The API requires the body be json.  
Your error (from PostgREST.org) is `An invalid request body was sent(e.g. an empty body or malformed JSON).`

**Reply by gflegg** [reply]
So, what needs to be changed from supabase crash course (and screen print shown) so that json will be correctly formed...?

**Reply by GaryAustin1** [reply]
I don't know anything about the crash course and why you need to use postman for it.
But in Postman if you want to send json data you have to select `raw` for the format and select json, I believe.

**Reply by steve-chavez** [reply]
@gflegg Make sure you're sending the `Content-Type: application/json` header.


(PostgREST does support the `application/x-www-form-urlencoded` type([doc](https://postgrest.org/en/stable/api.html#insertions)) but the happy path for the hosted version on Supabase is json)

**Reply by gflegg** [reply]
YES, THANKS  GARY - WHEN I SENT RAW JSON FORMATTED DATA MY POST SUCCEEDED WITH '201 CREATED'...

The crash course https://www.youtube.com/watch?v=7uKQBl9uZ00 at around 13 minutes instructed sending data as a x-www-form-encoded key value pair (with header Content-Type of application/json) which failed - as posted in and shown in the print screen of my question.

The need (reason for) using postman was to test and see that the api works for the supabase table...
